core/irCheck.ml

```diff
@@ -5,30 +5,85 @@ open Ir
 (* ERROR HANDLING*)
 
 let fail_on_ir_type_error = false
-exception IRTypeError of string
+
+(* Artificial "supertype" of all IR types,
+   so we can provide arbitrary IR fragments to the error handling functions *)
+type ir_snippet =
+  | STC    of tail_computation
+  | SVal   of value
+  | SSpec  of special
+  | SBind  of binding
+  | SBinds of binding list
+  | SProg  of program
+  | SNone
+
+let string_of_occurrence : ir_snippet -> string = function
+  | STC tc -> "\noccurring in IR tail computation:\n" ^ Ir.string_of_tail_computation tc
```

Comment:
I don't like inserting `\n` into strings. Is there no function or combinator suited to do just that?

Also, maybe align `->`?

Reply:
Yes, I agree it's ugly. We have a little pretty-printier in `pp.ml`, I'll give it a try.

Reply:
How do you like the following:


```
let string_of_occurrence : ir_snippet -> string =
  let nl = "\n" in
  function
  | STC tc    ->
     nl ^
     "occurring in IR tail computation:" ^
     nl ^
     Ir.string_of_tail_computation tc
  | ...
```


I didn't use the `pp` thing, it turned out to make things more complicated, rather than less.